disttask/framework/scheduler/scheduler.go

```diff
@@ -436,16 +449,37 @@ func (s *BaseScheduler) resetError() {
 	s.mu.handled = false
 }
 
-func (s *BaseScheduler) startSubtask(id int64) {
-	err := s.taskTable.StartSubtask(id)
+func (s *BaseScheduler) startSubtask(subtask *proto.Subtask) {
+	metrics.DecDistDDLSubTaskCnt(subtask)
+	metrics.EndDistDDLSubTask(subtask)
+	err := s.taskTable.StartSubtask(subtask.ID)
 	if err != nil {
 		s.onError(err)
 	}
+	subtask.State = proto.TaskStateRunning
+	metrics.IncDistDDLSubTaskCnt(subtask)
+	metrics.StartDistDDLSubTask(subtask)
 }
 
-func (s *BaseScheduler) updateSubtaskStateAndError(subtaskID int64, state string, subTaskErr error) {
-	err := s.taskTable.UpdateSubtaskStateAndError(subtaskID, state, subTaskErr)
+func (s *BaseScheduler) updateSubtaskStateAndError(subtask *proto.Subtask, state string, subTaskErr error) {
+	metrics.DecDistDDLSubTaskCnt(subtask)
+	metrics.EndDistDDLSubTask(subtask)
+	err := s.taskTable.UpdateSubtaskStateAndError(subtask.ID, state, subTaskErr)
 	if err != nil {
 		s.onError(err)
 	}
+	subtask.State = state
+	metrics.IncDistDDLSubTaskCnt(subtask)
+	metrics.StartDistDDLSubTask(subtask)
+}
+
+func (s *BaseScheduler) finishSubtask(subtask *proto.Subtask, subtaskMeta []byte) {
```

Comment:
Where is the finishSubtask method called?

Reply:
fixed in https://github.com/pingcap/tidb/pull/47175/commits/c230f370f49722a60cf611e30ce2e9f3fffce707